vslm/global_object_manager.go

```diff
@@ -0,0 +1,659 @@
+}
+
+func (this *GlobalObjectManager) Retrieve(ctx context.Context, id vim.ID) (*vim.VStorageObject, error) {
+	req := types.VslmRetrieveVStorageObject{
+		This: this.Reference(),
+		Id:   id,
+	}
+
+	res, err := methods.VslmRetrieveVStorageObject(ctx, this.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return &res.Returnval, nil
+}
+
+func (this *GlobalObjectManager) RetrieveState(ct context.Context, id vim.ID) (*vim.VStorageObjectStateInfo, error) {
+	req := types.VslmRetrieveVStorageObjectState{
+		This: this.Reference(),
+		Id:   id,
+	}
+
+	res, err := methods.VslmRetrieveVStorageObjectState(ct, this.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return &res.Returnval, nil
+}
+
+func (this *GlobalObjectManager) RetrieveAssociations(ct context.Context, ids []vim.ID) (
+	[]types.VslmVsoVStorageObjectAssociations, error) {
+	req := types.VslmRetrieveVStorageObjectAssociations{
+		This: this.Reference(),
+		Ids:  ids,
+	}
+
+	res, err := methods.VslmRetrieveVStorageObjectAssociations(ct, this.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return res.Returnval, nil
+}
+
+func (this *GlobalObjectManager) ListObjectsForSpec(ctx context.Context, query []types.VslmVsoVStorageObjectQuerySpec,
+	maxResult int32) (*types.VslmVsoVStorageObjectQueryResult, error) {
+	req := types.VslmListVStorageObjectForSpec{
+		This:      this.Reference(),
+		Query:     query,
+		MaxResult: maxResult,
+	}
+
+	res, err := methods.VslmListVStorageObjectForSpec(ctx, this.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return res.Returnval, nil
+}
+
+func (this *GlobalObjectManager) Clone(ctx context.Context, id vim.ID, spec vim.VslmCloneSpec) (*Task, error) {
+	req := types.VslmCloneVStorageObject_Task{
+		This: this.Reference(),
+		Id:   id,
+		Spec: spec,
+	}
+
+	res, err := methods.VslmCloneVStorageObject_Task(ctx, this.c, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewTask(this.c, res.Returnval), nil
+}
+
+func (this *GlobalObjectManager) Relocate(ct context.Context, id vim.ID, spec vim.VslmRelocateSpec) (*Task, error) {
+	req := types.VslmRelocateVStorageObject_Task{
+		This: this.Reference(),
+		Id:   id,
+		Spec: spec,
+	}
+
+	res, err := methods.VslmRelocateVStorageObject_Task(ct, this.c, &req)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return NewTask(this.c, res.Returnval), nil
+}
+
+func (this *GlobalObjectManager) SetControlFlags(ct context.Context, controlFlags []string) error {
+	req := types.VslmSetVStorageObjectControlFlags{
+		This:         this.Reference(),
+		ControlFlags: controlFlags,
+	}
+
+	_, err := methods.VslmSetVStorageObjectControlFlags(ct, this.c, &req)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (this *GlobalObjectManager) ClearControlFlags(ct context.Context) error {
+	req := types.VslmClearVStorageObjectControlFlags{
+		This: this.Reference(),
+	}
+
+	_, err := methods.VslmClearVStorageObjectControlFlags(ct, this.c, &req)
+	if err != nil {
```

Comment:
No need for this check, can just `return err` here.

Reply:
Fixed